extensions/sql-migration/src/service/serviceClient.ts

```diff
@@ -78,13 +78,22 @@ export class ServiceClient {
 	}
 
 	public async downloadBinaries(context: vscode.ExtensionContext, rawConfig: Buffer): Promise<string> {
-		const config = JSON.parse(rawConfig.toString());
-		config.installDirectory = path.join(context.extensionPath, config.installDirectory);
-		config.proxy = vscode.workspace.getConfiguration('http').get('proxy');
-		config.strictSSL = vscode.workspace.getConfiguration('http').get('proxyStrictSSL', true);
-		const serverdownloader = new ServerProvider(config);
-		serverdownloader.eventEmitter.onAny(this.generateHandleServerProviderEvent());
-		return serverdownloader.getOrDownloadServer();
+		try {
+			const config = JSON.parse(rawConfig.toString());
+			config.installDirectory = path.join(context.extensionPath, config.installDirectory);
+			config.proxy = vscode.workspace.getConfiguration('http').get('proxy');
+			config.strictSSL = vscode.workspace.getConfiguration('http').get('proxyStrictSSL', true);
+			const serverdownloader = new ServerProvider(config);
+			serverdownloader.eventEmitter.onAny(this.generateHandleServerProviderEvent());
+			return serverdownloader.getOrDownloadServer();
+		}
+		catch (error) {
+			const errorStr = localize('downloadingServiceFailed', "Failed to download binaries for {0}. Use the following link to troubleshoot: {1}", constants.serviceName, "https://learn.microsoft.com/en-us/azure/dms/known-issues-azure-sql-migration-azure-data-studio#azure-data-studio-limitations");
+			this.outputChannel.appendLine(errorStr);
+			await vscode.window.showErrorMessage(errorStr);
```

Comment:
Message boxes support markdown links, so you should at the very least do that. You could also have a button that will open up the URL when clicked.